### DISCUSSION TITLE
how can I connect min.io to a local KES server with autosigned certificate?

### DISCUSSION THREAD

**Discussion by jpchev** [minio/minio#14617] [Q&A] [answered]
I've set up a KES server using a key/certificate created with the following command

`kes tool identity new --server --key kes/kes_server.key --cert kes/kes_server.crt --ip "127.0.0.1" --dns localhost`

now when I start min.io with the following environment variables

```
export MINIO_KMS_KES_ENDPOINT=https://127.0.0.1:7373
export MINIO_KMS_KES_KEY_FILE=/home/centos/kes/app.key
export MINIO_KMS_KES_CERT_FILE=/home/centos/kes/app.cert
export MINIO_KMS_KES_KEY_NAME=my-app-key
```

I get the following error
`ERROR Unable to initialize a connection to KES as specified by the shell environment: Post "https://127.0.0.1:7373/v1/key/create/my-app-key": x509: certificate signed by unknown authority`

is there an option to make min.io skip the certificate validation?

**Comment by klauspost** [comment, marked as answer]
Follow the guide here: https://github.com/minio/kes/wiki/MinIO-Object-Storage

In particular see "Trust the KES Server Certificate".

**Reply by jpchev** [reply]
thanks for your answer
